examples/2fa-sms-login.example.ts

```diff
@@ -0,0 +1,47 @@
+import { IgApiClient } from '../src';
+import { get } from 'lodash';
+
+const readline = require('readline');
+const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
+
+(async () => {
+  // Initiate Instagram API client
+  const ig = new IgApiClient();
+  ig.state.generateDevice(process.env.IG_USERNAME);
+  ig.state.proxyUrl = process.env.IG_PROXY;
+
+  let loggedInUser;
+  try {
+    // Perform usual login
+    loggedInUser = await ig.account.login(process.env.IG_USERNAME, process.env.IG_PASSWORD);
+  } catch (e) {
+    // If 2FA is enabled, IgLoginTwoFactorRequiredError will be thrown
+    // Skip if another type of error is thrown
+    if (e.name !== 'IgLoginTwoFactorRequiredError') {
+      throw e;
+    }
+    // Check response contains identifier
+    const twoFactorIdentifier = get(e, 'response.body.two_factor_info.two_factor_identifier');
+    if (!twoFactorIdentifier) {
+      throw new Error('Unable to login, no 2fa identifier found');
+    }
+    // At this point a code should have been received via SMS
+    // Get SMS code from stdin
+    const code = await getStdinData('Sms code-> ');
```

Comment:
For consistency you may use inquirer like here: https://github.com/dilame/instagram-private-api/blob/a716332eb91c6a6986af058f0016b01aec4cbc91/examples/checkpoint.example.ts#L17

Reply:
Same here, wanted to avoid any dependencies